src/routers/posts.ts

```diff
@@ -22,25 +22,1077 @@ import {
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred.
+ *                 err:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.patch('/changeRange/:postID', auth, postRangeControl, changePostRange);
+
+/**
+ * @swagger
+ * /posts/changeActive/{postID}:
+ *   patch:
+ *     tags:
+ *       - Posts operations
+ *     summary: Toggle the active status of a post, Must be connected as admin
+ *     description: Toggles the `active` status of a post by its ID. Admins can change the status of any post, but the operation will not affect posts with a specific `superID` which is protected.
+ *     security:
+ *       - BearerAuth: []  # Indicates that the endpoint requires a bearer token for authentication
+ *     parameters:
+ *       - in: path
+ *         name: postID
+ *         required: true
+ *         schema:
+ *           type: string
+ *         description: The unique identifier of the post whose active status is being toggled.
+ *     responses:
+ *       200:
+ *         description: Success message with the updated post data.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 _id:
+ *                   type: string
+ *                   description: The unique identifier of the post.
+ *                 active:
+ *                   type: boolean
+ *                   description: The updated active status of the post.
+ *                 updatedAt:
+ *                   type: string
+ *                   format: date-time
+ *                   description: The timestamp of when the post was last updated.
+ *       401:
+ *         description: Unauthorized error if attempting to change the status of a protected post or if not authorized.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating why the status could not be changed.
+ *       500:
+ *         description: Internal server error
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 msg:
+ *                   type: string
+ *                   description: Error message indicating that an error occurred during the operation.
+ *                 err:
+ *                   type: object
+ *                   description: Details of the error encountered.
+ */
 router.patch('/changeActive/:postID', authAdmin, changeActiveStatus);
 
+/**
+ * @swagger
+ * /posts/{postID}:
+ *   delete:
+ *     tags:
+ *       - Posts operations
+ *     summary: Delete a post and its associated images , Must be connected as user
```

Comment:
summary should contains just -
"Must be connected as admin" OR
"Must be connected as user"